### ISSUE TITLE
an InvalidArgumentError occurred when use CarliniWagnerL2 to generate adversarial examples

### ISSUE THREAD

**Issue by lijinfeng0713** [tensorflow/cleverhans#588]
When I create a CarliniWagnerL2 instance, everything is ok
`cw = CarliniWagnerL2(session, model, 10, 0.3, False, 0.2, 10, 1000, True, 10, 0, 1, 10,
                         model.x.get_shape().as_list()[1:])`

but when I use "attack_batch()" function to generate adversarial examples, there occurred an InvalidArgumentError.
`adv_x = cw.attack_batch(mnist.test.images[0:100], mnist.test.labels[0:100])`

 Here is the detail info:

InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'x' with dtype float and shape [?,784]
	 [[Node: x = Placeholder[dtype=DT_FLOAT, shape=[?,784], _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
	 [[Node: logits/_31 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_142_logits", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

**Comment by sktzwhj** [comment]
How can you create the instance as the constructor only takes at most 5 arguments?

**Comment by npapernot** [comment]
My guess is that you are using the class that is in `attacks_tf.py` rather than the one in `attacks.py`. Can you try to make the change and tell us if that worked? You can find a tutorial here: https://github.com/tensorflow/cleverhans/blob/master/cleverhans_tutorials/mnist_tutorial_cw.py
